DOCSP-27478 Atlas search (#67)
mongodb/docs-csharp: source/fundamentals/builders.txt

To learn more about the Aggregation Pipeline, see the
:manual:`Aggregation Pipeline </core/aggregation-pipeline/>` server manual page.

Build an Atlas Search Query
---------------------------

The ``Search`` class provides a type-safe interface for creating a
:manual:`$search </reference/operator/aggregation/search/>` 
pipeline stage. 

> .. note:: Only Available on Atlas for MongoDB v4.2 and later

Reviewer: ```suggestion
.. note:: Only Available on Atlas for MongoDB v4.2 and Later
```

>    This aggregation pipeline operator is only available for collections hosted

Reviewer: Issue: Clarify operator + compound modifier + ambiguous "only"

```suggestion
   The ``$search`` aggregation-pipeline operator is available only for collections hosted
```

>    on :atlas:`MongoDB Atlas </>` clusters running v4.2 or later that are

Reviewer: ```suggestion
   on :atlas:`MongoDB Atlas </>` clusters running MongoDB v4.2 or later that are
```

   covered by an :atlas:`Atlas search index </reference/atlas-search/index-definitions/>`.
   Learn more about the required setup and the functionality of this operator
   from the :ref:`Atlas Search <fts-top-ref>` documentation.

The following example creates a pipeline stage that searches for any document
that has the word "yellow" in the ``Description`` field.

.. code-block:: csharp

   var result = flowerCollection.Aggregate()
       .Search(Builders<Flower>.Search.Text(f => f.description, "yellow"));

> You can also build a search query that searches using multiple operations. The
> following example searches for any document that meets the following criteria:

Reviewer: Sugg: Could sound a little cleaner? How about:

```suggestion
You can also combine multiple operations into a single search query. For example, 
the following code sample searches for any document that meets the following criteria:
```

> following example searches for any document that meets the following criteria:

Reviewer: Issue: Searches for any (one) document? Or all such documents?

Sugg: Clarify

- Contains the word "yellow" in the ``Description`` field
- Contains the word "Perennial" in the ``Category`` field
- Doesn't contain the word "blue" in the ``Description`` field

.. code-block:: csharp

   var result = flowerCollection.Aggregate()
       .Search(Builders<Flower>.Search.Compound()
             .Must(Builders<Flower>.Search.Text(f => f.Description, "yellow"),
                   Builders<Flower>.Search.Text(f => f.Category, "Perennial"))
             .MustNot(Builders<Flower>.Search.Text(f => f.Description, "blue"))
       );

> To learn more about the operations available to the ``$search`` pipeline stage,

Reviewer: ```suggestion
To learn more about the operations available in the ``$search`` pipeline stage,
```
